test/suite/wtbackup.py

```diff
@@ -144,6 +192,143 @@ def compare_backups(self, uri, base_dir_home, other_dir_home, suffix = None):
+                self.range_copy(newfile, offset, size, backup_incr_dir)
+                lens.append(size)
+        self.assertEqual(ret, wiredtiger.WT_NOTFOUND)
+        incr_c.close()
+        return lens
+
+    #
+    # Given a backup cursor, open a log cursor, and copy all log files that are not
+    # in the given log list. Return all the log files
+    #
+    def take_log_backup(self, bkup_c, backup_dir, orig_logs, log_cursor=None):
+        # Now open a duplicate backup cursor.
+        dupc = log_cursor
+        if log_cursor == None:
+            config = 'target=("log:")'
+            dupc = self.session.open_cursor(None, bkup_c, config)
+        dup_logs = []
+        while True:
+            ret = dupc.next()
+            if ret != 0:
+                break
+            newfile = dupc.get_key()
+            self.assertTrue("WiredTigerLog" in newfile)
+            sz = os.path.getsize(newfile)
+            if (newfile not in orig_logs):
+                self.pr('DUP: Copy from: ' + newfile + ' (' + str(sz) + ') to ' + backup_dir)
+                shutil.copy(newfile, backup_dir)
+            # Record all log files returned for later verification.
+            dup_logs.append(newfile)
+        self.assertEqual(ret, wiredtiger.WT_NOTFOUND)
+        if log_cursor == None:
+            dupc.close()
+        return dup_logs
+
+    #
+    # Open incremental backup cursor, with an id and iterate through all the files
+    # and perform incremental block copy for each of them. Returns the information about
+    # the backup files.
+    #
+    # Optional arguments:
+    # consolidate: Add consolidate option to the cursor
+    # ret_sizes: Flag to return sizes of files instead of default
+    #
+    def take_incr_backup(self, backup_incr_dir, id=0, consolidate=False, ret_sizes=False):
+        self.assertTrue(id > 0 or self.bkup_id > 0)
+        if id == 0:
+            id = self.bkup_id
+        # Open the backup data source for incremental backup.
+        config = 'incremental=(src_id="ID' +  str(id - 1) + '",this_id="ID' + str(id) + '"'
+        if consolidate:
+            config += ',consolidate=true'
+        config += ')'
+        self.pr(config)
+        bkup_c = self.session.open_cursor('backup:', None, config)
+
+        files_info = []
+        # We cannot use 'for newfile in bkup_c:' usage because backup cursors don't have
+        # values and adding in get_values returns ENOTSUP and causes the usage to fail.
+        # If that changes then this, and the use of the duplicate below can change.
+        while True:
+            ret = bkup_c.next()
+            if ret != 0:
+                break
+            newfile = bkup_c.get_key()
+            file_sizes = self.take_incr_backup_block(bkup_c, newfile, backup_incr_dir)
+            if ret_sizes:
+                files_info += file_sizes
+            else:
+                files_info.append(newfile)
```

Comment:
Having this function return two things, of different types, depending on a parameter, is hard to follow.
I think it is better to not have the `ret_sizes` parameter and always return multiple values (`return (file_names, file_sizes)`.  Caller does `(file_names, file_sizes) = take_incr_backup(...)` and you can choose to ignore info you don't need.

Reply:
done!